### PR TITLE
64 implement facebook and google auth methods in flutter

### LINKED ISSUE #64: Implement Facebook and Google auth methods in Flutter
**Short description:**
Imagine and implement a way to login to the server by using Facebook or Google auth ways.

**Describe the solution you'd like**
Maybe launch a Navigator page and parse the content after the answer ?
Not much other ideas for the moment...

### PR DESCRIPTION
**Pull request by LucasTesnier**
# Purpose

The purpose of this pull request is to add the Google authentication way.
Fixes #64 

## Added features

* You can now login with Google from Flutter app

## Removed features

* Google api auth ways have been removed.

### CODE REVIEW

File: application/app_code/lib/pages/login/login_state.dart [hunk 1]
```diff
@@ -22,6 +23,49 @@ class LoginPageState extends State<LoginPage> {
...
+              UserData.fromJson(jsonDecode(response.body)['data']);
+          return 'Login succeed !';
+        } catch (err) {
+          return 'Invalid token... Please retry';
+        }
+      } else {
+        try {
+          return jsonDecode(response.body)['message'];
+        } catch (err) {
+          return 'Google Auth temporaly desactivated.';
```
Comment (tbellicha) [inline application/app_code/lib/pages/login/login_state.dart]: `temporaly` -> `temporarily`

File: application/app_code/lib/pages/login/login_state.dart [hunk 2]
```diff
@@ -22,6 +23,49 @@ class LoginPageState extends State<LoginPage> {
...
+        }
+      }
+    } catch (e) {
+      debugPrint(e.toString());
+      return 'Google Auth temporaly desactivated.';
```
Comment (tbellicha) [inline application/app_code/lib/pages/login/login_state.dart]: `temporaly` -> `temporarily`

**Review by tbellicha — CHANGES_REQUESTED** [review]

**Review by TomDUVAL-MAHE — DISMISSED** [review]
NTR

File: application/app_code/lib/pages/login/login_state.dart (hunk 1 above)
Comment (LucasTesnier) [reply application/app_code/lib/pages/login/login_state.dart]: done

File: application/app_code/lib/pages/login/login_state.dart (hunk 2 above)
Comment (LucasTesnier) [reply application/app_code/lib/pages/login/login_state.dart]: done

**Review by TomDUVAL-MAHE — APPROVED** [review]

**Review by ZQUEMA — APPROVED** [review]

**Review by tbellicha — APPROVED** [review]

**Pull request merged**